src/typecode/contenttype.py

```diff
@@ -287,6 +289,11 @@ def is_archive(self):
         """
         # FIXME: we should use extracode archive detection
         # TODO: also treat file systems as archives
+        #adding extract code detector
+        can_extract(location)
+        if can_extract(location) == True:
+            extract(location, kinds=extractcode.default_kinds, recurse=False)
```

Comment:
Are you seriously going to extract all archives just when doing a file type check? Do not extract here!

Reply:
So may be the only thing to do is:
`return can_extract(location)`

Reply:
Thanks alot.
Can you guide me about how to write new unit tests?